### PR TITLE
Updating Dockerfile to fit best practices

### PR DESCRIPTION
**Pull request by timgent**
- Use ruby-installer as it is a little lighter weight than rvm (no point in rvm as people shouldn't be changing ruby versions within their app)
- Has onbuild commands for running yum commands
- Has onbuild commands for copying app to /app and
- Explicitly says in readme to use the app user that is created
- Updated to use latest stable ruby version

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
     useradd -r -g app app -d /app && \
     mkdir -p /app && \
     chown -R app:app /app
 
-ENV GEM_PATH=/usr/local/rvm/gems/ruby-${RUBY_VERSION}:/usr/local/rvm/gems/ruby-${RUBY_VERSION}@global \
-    RUBY_VERSION=ruby-${RUBY_VERSION} \
-    GEM_HOME=/usr/local/rvm/gems/ruby-${RUBY_VERSION} \
-    PATH=${PATH}:/usr/local/rvm/rubies/ruby-${RUBY_VERSION}/bin:/usr/local/rvm/gems/ruby-${RUBY_VERSION}/bin:/usr/local/rvm/gems/ruby-${RUBY_VERSION}@global/bin
-
 WORKDIR /app
```
Comment (Joseph-Irving) [inline Dockerfile:36]: You should set USER in the dockerfile

**Review by Joseph-Irving — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Joseph-Irving) [reply Dockerfile:36]: Though I noticed you suggest people to use this later on when building on top of this image. Might be worth baking it in anyway just for good practice

Comment (timgent) [reply Dockerfile:36]: @Joseph-Irving you can't set the user in this Dockerfile as for the yum update etc to work with the onbuild command it'll still need to be root

Comment (Joseph-Irving) [reply Dockerfile:36]: ok, good point

**Review by Joseph-Irving — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,32 +1,36 @@
 FROM quay.io/ukhomeofficedigital/centos-base
 
-ENV RUBY_VERSION 2.3.0
+ENV RUBY_VERSION 2.3.3
```
Comment (marcinc) [inline Dockerfile:3]: Do you still need this, given `ruby-install --system ruby 2.2.6` below?

Comment (timgent) [reply Dockerfile:3]: Hah erm nope good point thanks

**Pull request merged**